tests/testsuite/resolve.rs

```diff
@@ -185,6 +190,97 @@ fn loc_names(names: &[(&'static str, &'static str)]) -> Vec<PackageId> {
         .collect()
 }
 
+/// This generates a random registry index.
+/// Unlike vec((Name, Ver, vec((Name, VerRq), ..), ..)
+/// This strategy has a high probability of having valid dependencies
+fn registry_strategy() -> impl Strategy<Value=Vec<Summary>> {
+    const VALID_NAME_STRATEGY: &str = "[A-Za-z_-][A-Za-z0-9_-]*";
```

Comment:
Since it sounds like runtime performance is a problem right now, I'd suggest [precompiling](https://altsysrq.github.io/rustdoc/proptest/0.8.5/proptest/string/fn.string_regex.html) the regex so it doesn't get reparsed every time.